### ISSUE TITLE
Intent to change signature of `context_sensitive_executable_path`

### ISSUE THREAD

**Issue by kaste** [SublimeLinter/SublimeLinter#1362]
`context_sensitive_executable_path` had a polymorphic signature in that
it also allowed the full `cmd` to be passed in. However all
implementations then only extracted `cmd[0]` and acted upon that.

Esp. for the new `which_executable` we want a simpler signature, so that
the user can just call e.g. `self.which_executable('mypy')`. We, of
course, could transform this internally to `['mypy']`, pass it down,
and transform it back to `'mypy'` in every `csep` implementation but
that seems unnecessary. 

So the question is, is it sufficient to compute the executable from one
`str` (like all our implementations), or does the executable depend on the 
full `cmd`.

**Comment by kaste** [comment]
Of course, ideally we could just call something like self.get_executable() bc the 'mypy' can be inferred.

**Comment by FichteFoll** [comment]
The use case would be an executable setting override from the user like `['python', '-m', 'mypy']`, correct?

**Comment by kaste** [comment]
I don't quite understand what you mean. It's already implemented in #1363 though. I think this should be mostly internal. The *call* signature changes from `Union[List[str], str]` to just `str`. 

If a user wants to know which executable is configured, you just ask `get_executable()` which is a typical getter from `None` to `Optional[List[str]]`

**Comment by FichteFoll** [comment]
I'm just not sure currently when user overrides (of `executable`) are accounted for and when the `csep` is called, i.e. which is first. It's been a while since I looked at the code. I'm thinking `csep` would be called beforehand, however.
I don't have a particular use case in mind, either.
